### ISSUE TITLE
Vulnerabilities in the aws-appmesh-envoy image (v1.21.1.2-prod)

### ISSUE THREAD

**Issue by dgr237** [aws/aws-app-mesh-roadmap#405]
We have recently scanned the above image and it has identified the following vulnerabilities:

CVE-2022-23218 (https://nvd.nist.gov/vuln/detail/CVE-2022-23218)
CVE-2022-23219 (https://nvd.nist.gov/vuln/detail/CVE-2022-23219)

Would it be possible to get an image which has these vulnerabilities fixed?

**Comment by suniltheta** [comment]
Hi @dgr237 these AL2 associated CVE will be fixed the next time we do an envoy image release for v1.22

To mitigate these issues for the time being would it be possible for you to generate a custom image? You can run the below command to get a custom image that will not hit the vulnerabilities.

```
cat << EOF > Dockerfile
FROM public.ecr.aws/appmesh/aws-appmesh-envoy:v1.21.1.2-prod
RUN yum -y update && \
    yum clean all && \
    rm -rf /var/cache/yum
EOF

docker build -f Dockerfile -t aws-appmesh-envoy:v1.21.1.2-prod-cve-fix .
```

> Note: Pick the envoy image that you are currently using instead of `public.ecr.aws/appmesh/aws-appmesh-envoy:v1.21.1.2-prod` below.

**Comment by suniltheta** [comment]
Hi, we have released a new version of AppMesh envoy image `v1.22.0` https://github.com/aws/aws-app-mesh-roadmap/issues/407

The CVE's mentioned above should not be seen. 

If you do not want to switch to `v1.22` yet, we are also planning to do another minor version release of `v1.21.2` in a week.

**Comment by dgr237** [comment]
I can confirm the 1.22 image has no reported vulnerabilities.

David

**Comment by suniltheta** [comment]
Closing this issue since Envoy v1.21.2 https://github.com/aws/aws-app-mesh-roadmap/issues/412 & Envoy v1.22.0 https://github.com/aws/aws-app-mesh-roadmap/issues/407 released images shouldn't contain these CVEs.

Although we are not sure about v1.21.2 as it may hit a new CVEs in future for unused packages present in the image as it is based on Amazon Linux 2 base. For v1.22.0 on the other hand is build as a distroless image which reduces our vulnerability exposure in unused packages present in the image. For this reason we advise all customers to move to v1.22.0+ versions asap.

**Comment by dgr237** [comment]
Unfortunately we would need to onboard the image into our in house
repository in order to build the image from your base image. We are blocked
from onboarding the image due to the vulnerability. Bit of a catch 22
situation.

Regards

David

**Comment by suniltheta** [comment]
only workaround is to wait for the new image release and onboard them.
We just released v1.23.1.0-prod that is built as distroless (scratch base) docker image. You should not see any AL2 vulnerabilities on that scratch image.

We include a Go binary built with golang v1.19.1 in to the image. There is a potential that we would hit Go CVEs in the future.
